contracts/src/lib/SupplySourceHelper.sol

```diff
@@ -46,12 +47,19 @@ library SupplySourceHelper {
     }
 
     /// @notice Locks the specified amount sent by the msg.sender into custody.
-    function lock(SupplySource memory supplySource, uint256 value) internal {
+    function lock(SupplySource memory supplySource, uint256 value) internal returns (uint256) {
         if (supplySource.kind == SupplyKind.ERC20) {
             IERC20 token = IERC20(supplySource.tokenAddress);
+            uint256 initialBalance = token.balanceOf(address(this));
             token.safeTransferFrom({from: msg.sender, to: address(this), value: value});
+            uint256 finalBalance = token.balanceOf(address(this));
+            if (!(finalBalance > initialBalance)) {
+                revert NoBalanceIncrease();
+            }
+            return finalBalance - initialBalance;
```

Comment:
Can you leave a comment here that no safe arithmetic is necessary because no underflow is possible due to the check above?

Reply:
added comment here 320cadd9